src/coprocessor/dag/dag.rs

```diff
@@ -55,21 +63,41 @@ impl DAGContext {
             req_ctx.isolation_level,
             req_ctx.fill_cache,
         );
+        let cache_key = format!("{:?}, {:?}", ranges, req.get_executors());
```

Comment:
The key is too long, could we hash it first?

Reply:
I think yes, but which hash function we should use? SHA1, SHA256? @siddontang as we discussed use hash function we may hit hash collision, if it is acceptable we can change it to hash function.

Reply:
It seems you should also consider other arguments in dag request like 
```
       // Transaction start timestamp.
	StartTs uint64 
	// time zone offset in seconds
	TimeZoneOffset int64 
	// flags are used to store flags that change the execution mode, it contains:
	// 	ignore_truncate = 1
	// 		truncate error should be ignore if set.
	// 	truncate_as_warning = 1 << 1
	// 		when ignored_truncate is not set, return warning instead of error if this flag is set.
	// 	...
	// 	add more when needed.
	Flags uint64 
	// It represents which columns we should output.
	OutputOffsets    []uint32 
```
Since `StartTS` will change frequently,  and only the request with a bigger startTS can use the cache according to the current implementation. We can cache the biggest `startTS` instead of encoding it in each cache key. 
But for other params, all of them seems should be encoded to the cache key.

Reply:
Meanwhile,  `ranges` should be encoded into the cache key too, since it may change for the same dag request.

Reply:
@AndreMouche I think OutputOffsets and Flags should include in cache key, but StartTS and TimeZoneOffset is not need. If other transaction modified data, the related caches are flushed. So different transaction run in same query should get same result when no data changed.